Tools/autotest/arduplane.py

```diff
@@ -1765,6 +1765,87 @@ def fly_terrain_mission(self):
+        self.set_parameter("EK3_AFFINITY", 15) # enable affinity for all sensors
+        self.set_parameter("EK3_IMU_MASK", 3) # use only 2 IMUs
+        self.set_parameter("GPS_TYPE2", 1)
+        self.set_parameter("SIM_GPS2_DISABLE", 0)
+        self.set_parameter("SIM_BARO2_DISABL", 0)
+        self.set_parameter("SIM_BARO_COUNT", 2)
+        self.set_parameter("ARSPD2_TYPE", 2)
+        self.set_parameter("ARSPD2_USE", 1)
+        self.set_parameter("ARSPD2_PIN", 2)
+
+        # some parameters need reboot to take effect
+        self.reboot_sitl()
+        
+        # get flying
+        self.takeoff(alt=25)
+        self.change_mode('CIRCLE')
+
+        self.context_push()
+        ex = None
+        try:
+            self.start_subtest("Checking Lane Switching trigger from all sensors")
+
+            # accelerometer
+            self.progress("ACCELEROMETER: Applying offset to z - axis")
+            self.context_push()
+            self.wait_statustext(text="lane switch", timeout=3, the_function=self.set_parameter("INS_ACCOFFS_Z", 5))
+            self.context_pop()
+            self.wait_heading(0, accuracy=10, timeout=60)
+            self.wait_heading(180, accuracy=10, timeout=60)
+
+            self.progress("GPS: Applying glitch to 2nd GPS")
+            self.context_push()
+            self.wait_statustext(text="lane switch", timeout=3, the_function=self.set_parameter("SIM_GPS2_GLTCH_X", 1))
+            self.context_pop()
+            self.wait_heading(0, accuracy=10, timeout=60)
+            self.wait_heading(180, accuracy=10, timeout=60)
+
+            self.progress("BAROMETER: Applying noise to 1st Barometer")
+            self.context_push()
+            self.wait_statustext(text="lane switch", timeout=3, the_function=self.set_parameter("SIM_BARO_RND", 100))
+            self.context_pop()
+            self.wait_heading(0, accuracy=10, timeout=60)
+            self.wait_heading(180, accuracy=10, timeout=60)
+            
+            self.progress("AIRSPEED: Applying offset to 2nd Airspeed Sensor")
+            self.context_push()
+            self.wait_statustext(text="lane switch", timeout=3, the_function=self.set_parameter("SIM_ARSPD2_OFS", 1500))
+            self.context_pop()
+            self.wait_heading(0, accuracy=10, timeout=60)
+            self.wait_heading(180, accuracy=10, timeout=60)
+
+            self.progress("MAGNETOMETER: Applying offset to 1st Compass")
+            self.context_push()
+            self.wait_statustext(text="lane switch", timeout=3, the_function=self.set_parameter("SIM_MAG_OFS_X", 150))
+            self.context_pop()
+            self.wait_heading(0, accuracy=10, timeout=60)
+            self.wait_heading(180, accuracy=10, timeout=60)
+
+            self.progress("Gyroscope: Applying offset to 2nd Gyroscope")
+            self.context_push()
+            self.wait_statustext(text="lane switch", timeout=10, the_function=self.set_parameter("INS_GYR2OFFS_Y", 1))
```

Comment:
we need to parse these in a more detailed fashion, so we check it does the correct lane change, not just that it changed lane at all.

Reply:
I don't really know what you are logging, but you should be able to use something similar to https://github.com/ArduPilot/ardupilot/blob/master/Tools/autotest/common.py#L5212 to look into the current log and search for the right message!

Reply:
parsing the string seems fine now, we can check the XKF4.PI (EKF3 Primary Core Index) but probably would also have to cross check with the time-stamp and might over complicate things. Will look into this option if the string parsing fails anywhere.